tools/kolla-images.py

```diff
@@ -0,0 +1,375 @@
+KollaImageTags = Dict[str, Dict[str, str]]
+
+# Maps a Kolla image to a list of containers that use the image.
+IMAGE_TO_CONTAINERS_EXCEPTIONS: Dict[str, List[str]] = {
+    "haproxy": [
+        "glance_tls_proxy",
+        "neutron_tls_proxy",
+    ],
+    "mariadb-server": [
+        "mariadb",
+        "mariabackup",
+    ],
+    "neutron-eswitchd": [
+        "neutron_mlnx_agent",
+    ],
+    "neutron-metadata-agent": [
+        "neutron_metadata_agent",
+        "neutron_ovn_metadata_agent",
+    ],
+    "nova-conductor": [
+        "nova_super_conductor",
+        "nova_conductor",
+    ],
+    "prometheus-v2-server": [
+        "prometheus_server",
+    ],
+}
+
+# Maps a container to the parent tag variable in the hierarchy.
+CONTAINER_TO_PREFIX_VAR_EXCEPTIONS: Dict[str, str] = {
+    "cron": "common",
+    "fluentd": "common",
+    "glance_tls_proxy": "haproxy",
+    "hacluster_corosync": "openstack",
+    "hacluster_pacemaker": "openstack",
+    "hacluster_pacemaker_remote": "openstack",
+    "heat_api_cfn": "heat",
+    "ironic_neutron_agent": "neutron",
+    "kolla_toolbox": "common",
+    "mariabackup": "mariadb",
+    "neutron_eswitchd": "neutron_mlnx_agent",
+    "neutron_tls_proxy": "haproxy",
+    "nova_compute_ironic": "nova",
+    "redis_sentinel": "openstack",
+    "swift_object_expirer": "swift",
+    "tgtd": "iscsi",
+}
+
+# List of supported base distributions and versions.
+SUPPORTED_BASE_DISTROS = [
+    "rocky",
+    "ubuntu",
+]
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--base-distros", default=",".join(SUPPORTED_BASE_DISTROS), choices=SUPPORTED_BASE_DISTROS)
+    subparsers = parser.add_subparsers(dest="command", required=True)
+
+    subparser = subparsers.add_parser("check-hierarchy", help="Check tag variable hierarchy against kolla-ansible")
+    subparser.add_argument("--kolla-ansible-path", required=True, help="Path to kolla-ansible repostory checked out to correct branch")
+
+    subparser = subparsers.add_parser("check-tags", help="Check specified tags for each image exist in the Ark registry")
+    subparser.add_argument("--registry", required=True, help="Hostname of container image registry")
+    subparser.add_argument("--namespace", required=True, help="Namespace in container image registry")
+
+    subparsers.add_parser("list-containers", help="List supported containers based on pulp.yml")
+
+    subparsers.add_parser("list-images", help="List supported images based on pulp.yml")
+
+    subparsers.add_parser("list-tags", help="List tags for each image based on kolla-image-tags.yml")
+
+    subparsers.add_parser("list-tag-vars", help="List Kolla Ansible tag variables")
+
+    return parser.parse_args()
+
+def get_distro_regex_map(openstack_release):
+    distro_to_regex = {
+        "rocky": re.compile(f"^{openstack_release}-rocky-9-[\d]{{8}}T[\d]{{6}}$"),
```

Comment:
Not sure if this is the cleanest solution but there is mismatch between the kolla_base_distro and the tag we use for the docker image (which contains the release version)...